### DISCUSSION TITLE
How to use Spacy 3.0.0 type hints with mypy

### DISCUSSION THREAD

**Discussion by mscheifer** [explosion/spaCy#7447] [Help: Best practices]
The [release notes for spacy 3.0.0](https://spacy.io/usage/v3#features-types) say that it now supports type hints:
> spaCy’s user-facing API that’s implemented in pure Python (as opposed to Cython) now comes with type hints

We check types in our builds with mypy. When I try to run it while importing spacy I get this error:
`Skipping analyzing 'spacy': found module but no type hints or library stubs`. After digging around I can't seem to find 1. a `py.typed` file in the spacy project. 2. A separate stubs package or 3. A stub entry in `typeshed`.

What's the intended way to make use of the type hints included in v3.0 ?

Thanks!

**Comment by adrianeboyd** [comment]
Hmm, it looks like we're just missing `py.typed` in spacy? I see it in thinc, but not in spacy.

**Reply by mscheifer** [reply]
Yeah I think so. I see you put up a PR: https://github.com/explosion/spaCy/pull/7451. Thanks!

**Comment by mscheifer** [comment]
This is working fine with 3.0.6
